RFS/src/main/java/com/rfs/tracing/RfsContexts.java

```diff
@@ -0,0 +1,453 @@
+        }
+
+        @Override
+        public String getActivityName() { return ACTIVITY_NAME; }
+
+        @Override
+        public IScopedInstrumentationAttributes getEnclosingScope() {
+            return null;
+        }
+
+        public static class MetricInstruments extends CommonScopedMetricInstruments {
+            private MetricInstruments(Meter meter, String activityName) {
+                super(meter, activityName);
+            }
+        }
+
+        public static @NonNull MetricInstruments makeMetrics(Meter meter) {
+            return new MetricInstruments(meter, ACTIVITY_NAME);
+        }
+
+        @Override
+        public MetricInstruments getMetrics() {
+            return getRootInstrumentationScope().createIndexInstruments;
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createCheckRequestContext() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    "CreateIndexContext.createCheckRequestContext");
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createPutContext() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    "CreateIndexContext.createPutContext");
+        }
+    }
+
+    public static class DocumentReindexContext
+            extends BaseSpanContext<RootRfsContext>
+            implements IRfsContexts.IDocumentReindexContext {
+
+        protected DocumentReindexContext(RootRfsContext rootScope) {
+            super(rootScope);
+            initializeSpan(rootScope);
+        }
+
+        @Override
+        public String getActivityName() { return ACTIVITY_NAME; }
+
+        @Override
+        public IScopedInstrumentationAttributes getEnclosingScope() {
+            return null;
+        }
+
+        public static class MetricInstruments extends CommonScopedMetricInstruments {
+            private MetricInstruments(Meter meter, String activityName) {
+                super(meter, activityName);
+            }
+        }
+
+        public static @NonNull MetricInstruments makeMetrics(Meter meter) {
+            return new MetricInstruments(meter, ACTIVITY_NAME);
+        }
+
+        @Override
+        public MetricInstruments getMetrics() {
+            return getRootInstrumentationScope().documentReindexInstruments;
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createBulkRequest() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    "DocumentReindexContext.createBulkRequest");
+        }
+
+        @Override
+        public IRfsContexts.IRequestContext createRefreshContext() {
+            return new GenericRequestContext(rootInstrumentationScope, this,
+                    "DocumentReindexContext.createRefreshContext");
+        }
+
+    }
+
+    public static class WorkingStateContext
+            implements IRfsContexts.IWorkingStateContext {
+        @Getter
+        @NonNull
```

Comment:
As a project go all in or out on these annotations.  Want to raise an issue/ticket on this?

Reply:
Lets revisit this later, we've got many usages no need to stick on this instance here.